Added sound interactions like error, success and click to about 70% of button interactions

outercloudstudio opened the pull request:
I edited where there was an on click (mostly in the scripts under the windows folder) I added one line of code that would play a sound file depending on the type of action. The sounds are stored in a audio file under the public folder and are CC0 licensed and from Kenney.nl.

I was motivated to make this change because I saw a recommendation for some audio in bridge [(this message)](https://discord.com/channels/602097536404160523/631516486644793344/841959372007276546)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

- [x] My code follows the code style of this project.
- [x] I have tested my changes and confirmed that they are working

solvedDev commented:
Thanks a lot for this 👏

Please see my comments on our Discord and don't hesitate to ask me further questions :)

https://discord.com/channels/602097536404160523/735256594769641482/844296469388197898

outercloudstudio commented:
I created an audio manager class that is created when the app class is created. Any script that wants to play a sound just has to call `App.audioManager.playAudio("audio name", audio volume)`. Also I added the sounds to the sidebar and controls class. The audio manager has a master volume setting which can be changed with a setting in the settings. I added a en localization for the audio settings too.

I made this change because making an audio manager class was suggested by soloDev and also to add a bit more functionality to audio.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)

- [ x] My code follows the code style of this project.
- [ x] I have tested my changes and confirmed that they are working

solvedDev reviewed (commented):
Looks great! Just a few further comments:
- The mute setting doesn't seem to be implemented just yet, is that intended? I can handle that once the PR is merged :)
- Can you remove the old comments "// new Audio(...)" and maybe put something like "// TODO: Add sound" instead?

outercloudstudio commented:
The mute setting plays a sound when you turn turn it off, then all other sounds should be muted. I might have made the toggle called mute but the volume mutes when the toggle is off.

outercloudstudio commented:
This is just a small fix removing some commented code and fixing the mute button.

solvedDev reviewed (approved):
LGTM, will merge the PR soon to ship audio in the next alpha

outercloudstudio commented:
It seemed to work on my end. It just failed to boot? Any crash logs or something? I'll see If I can find what's causing it.

solvedDev commented:
![Bildschirmfoto 2021-05-21 um 18 21 52](https://user-images.githubusercontent.com/33347616/119168671-71df8480-ba61-11eb-98e8-936e44e9f751.png)

outercloudstudio commented:
How could I reproduce this error?

solvedDev commented:
I didn't do anything special... I've just checked out your PR and started the dev server (`npm run dev`).
Maybe try giving the dev server a full restart?

outercloudstudio commented:
This should allow it to work. The audio manager script is there but it is not instanced or accessed for now. I'll see what is causing the error and then I'll add back the sound functionality. All I did was remove any lines of code calling the audio manager to play a sound. I also removed the static instance from app.ts.

solvedDev commented:
You might need to restart the dev server every time you add back a sound. Pretty annoying but that's how webpack works :/

solvedDev commented on src/components/Windows/Common/Dropdown/DropdownWindow.ts:
```diff
@@ -1,5 +1,7 @@
 import { BaseWindow } from '../../BaseWindow'
 import DropdownWindowComponent from './Dropdown.vue'
+import { App } from '/@/App'
+import { AudioManager } from '/@/components/Audio/AudioManager'
```
Is this import intentional?

solvedDev commented on src/components/Windows/Layout/Sidebar.ts:
```diff
@@ -202,6 +203,8 @@ export class Sidebar {
 	}
 	set selected(val) {
+		if (val) {
+		}
```
Empty if-statement here

solvedDev commented on src/components/Windows/Settings/setupSettings.ts:
```diff
@@ -218,6 +218,24 @@ export async function setupSettings(settings: SettingsWindow) {
+			default: false,
+			onChange: async (val) => {
+				if (val) {
+					App.audioManager.masterVolume = 0
```
The access of the audioManager now throws an error. I need to change this later anyways, you can uncomment the two lines for now.
![Bildschirmfoto 2021-05-21 um 18 48 10](https://user-images.githubusercontent.com/33347616/119171615-19aa8180-ba65-11eb-92d2-772096523988.png)

outercloudstudio commented:
Sorry, I forgot about that code. I commented it out for now.

outercloudstudio commented:
This should be the last commit that fixes the errors. Sorry about the hassle.

solvedDev commented:
No worries! LGTM
